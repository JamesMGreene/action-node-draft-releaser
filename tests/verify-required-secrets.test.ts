import path from 'path';
import verifyRequiredSecrets from '../src/verify-required-secrets';

describe('verifyRequiredSecrets', () => {
  afterEach(() => {
    delete process.env.GITHUB_EVENT_NAME;
    delete process.env.GITHUB_REF;
    delete process.env.GITHUB_REPOSITORY;
    delete process.env.GITHUB_SHA;
    delete process.env.GITHUB_TOKEN;
    delete process.env.GITHUB_WORKSPACE;
  });

  it('should not throw if all required secrets have values', () => {
    process.env.GITHUB_EVENT_NAME = 'push';
    process.env.GITHUB_REF = 'refs/heads/master';
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = 'deadbeefcafe1c6e148a9be56893bed50abadbed';
    process.env.GITHUB_TOKEN = 'testToken123';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    expect(verifyRequiredSecrets()).toBe(undefined);
  });

  it('should throw if none of the required secrets exist', () => {
    expect(verifyRequiredSecrets).toThrowErrorMatchingSnapshot();
  });

  it('should throw if all of the required secrets are missing values', () => {
    process.env.GITHUB_EVENT_NAME = '';
    process.env.GITHUB_REF = '';
    process.env.GITHUB_REPOSITORY = '';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_TOKEN = '';
    process.env.GITHUB_WORKSPACE = '';

    expect(verifyRequiredSecrets).toThrowErrorMatchingSnapshot();
  });

  it('should throw if some of the required secrets are missing values', () => {
    process.env.GITHUB_REPOSITORY = 'JamesMGreene/test-repo';
    process.env.GITHUB_SHA = '';
    process.env.GITHUB_TOKEN = 'testToken123';
    process.env.GITHUB_WORKSPACE = path.join(__dirname, 'fixtures');

    expect(verifyRequiredSecrets).toThrowErrorMatchingSnapshot();
  });
});
