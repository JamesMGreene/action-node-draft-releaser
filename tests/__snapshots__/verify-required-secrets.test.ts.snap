// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifyRequiredSecrets should throw if all of the required secrets are missing values 1`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_TOKEN
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredSecrets should throw if none of the required secrets exist 1`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_REPOSITORY
- GITHUB_SHA
- GITHUB_TOKEN
- GITHUB_WORKSPACE"
`;

exports[`verifyRequiredSecrets should throw if some of the required secrets are missing values 1`] = `
"The following secrets are required for this GitHub Action to run:
- GITHUB_EVENT_NAME
- GITHUB_REF
- GITHUB_SHA"
`;
